File: lemur-app/src/main/java/io/github/malczuuu/lemur/app/core/player/PlayerMapper.java
package io.github.malczuuu.lemur.app.core.player;

import io.github.malczuuu.lemur.app.domain.player.Player;

class PlayerMapper {

  PlayerItem toPlayerItem(Player player) {
    return new PlayerItem(
        player.getId().getValue(), player.getName(), player.getRating(), player.getStatus());
  }

  PlayerDetails toPlayerDetails(Player player) {
    return new PlayerDetails(
        player.getId().getValue(),
        player.getName(),
        player.getRating(),
        player.getStatus(),
        player.getCreatedDate().orElse(null),
        player.getLastModifiedDate().orElse(null),
        player.getVersion());
  }
}

File: lemur-app/src/main/kotlin/io/github/malczuuu/lemur/app/core/player/PlayerMapper.kt
package io.github.malczuuu.lemur.app.core.player

import io.github.malczuuu.lemur.app.domain.player.Player

internal class PlayerMapper {

    fun toPlayerItem(player: Player): PlayerItem =
        PlayerItem(id = player.id.value, name = player.name, rating = player.rating, status = player.status)

    fun toPlayerDetails(player: Player): PlayerDetails = PlayerDetails(
        id = player.id.value,
        name = player.name,
        rating = player.rating,
        status = player.status,
        createdDate = player.createdDate,
        lastModifiedDate = player.lastModifiedDate,
        version = player.version,
    )
}
